controllers/controller/workspacerouting/solvers/solver.go

```diff
@@ -98,9 +98,6 @@ func (_ *SolverGetter) HasSolver(routingClass controllerv1alpha1.WorkspaceRoutin
 }
 
 func (_ *SolverGetter) GetSolver(client client.Client, routingClass controllerv1alpha1.WorkspaceRoutingClass) (RoutingSolver, error) {
-	if routingClass == "" {
-		routingClass = controllerv1alpha1.WorkspaceRoutingClass(config.ControllerCfg.GetDefaultRoutingClass())
```

Comment:
If we don't support it, then used config `GetDefaultRoutingClass()` and related should be removed.

Or, instead defaulting can be moved to other level, let's say DevWorkpace controller, can update CR with default routing if it's missing.

I'm not sure it's really needed feature, but during development it was handy to switch between basic for k8s, and openshift-oauth for OpenShift, and all examples worked without change.

@amisevsk WDYT?

Reply:
I'm fine removing empty routingClass (or moving the defaulting into the main controller) but I also don't see the need for the change necessarily; I don't think DWO should be configured to use a default routingClass that it does not support, as that could lead to confusion (and also make integrating multiple dependants of DWO tricky).

I think this PR does what you suggest though, in still supporting devworkspaces without a routingClass, by setting the routingClass in the routing provision function.

Reply:
Ah, I must scroll down :-D

Actually I did but as I see not, did not read carefully enough